beard/clustering/wrappers.py

```diff
@@ -148,7 +169,7 @@ def fit(self, X, y=None):
         if y is not None:
             y_arr = np.array(y)
             all_y_neg = y_arr.sum() == len(y_arr) * -1
-            ground_truth = y is not None and not all_y_neg
+            ground_truth = not all_y_neg
         else:
             ground_truth = False
```

Comment:
Or even better, the whole if-statement with `ground_truth = (y is not None) and np.any(np.array(y) != -1)